web/frontend/tests/mocks/evoting.ts

```diff
@@ -0,0 +1,96 @@
+import Worker0 from './../json/api/proxies/dela-worker-0.json';
+import Worker1 from './../json/api/proxies/dela-worker-1.json';
+import Worker2 from './../json/api/proxies/dela-worker-2.json';
+import Worker3 from './../json/api/proxies/dela-worker-3.json';
+
+export const FORMID = 'b63bcb854121051f2d8cff04bf0ac9b524b534b704509a16a423448bde3321b4';
+
+export async function mockForms(page: page, empty: boolean = true) {
+  // clear current mock
+  await page.unroute(`${process.env.DELA_PROXY_URL}/evoting/forms`);
+  await page.route(`${process.env.DELA_PROXY_URL}/evoting/forms`, async (route) => {
+    if (route.request().method() === 'OPTIONS') {
+      await route.fulfill({
+        status: 200,
+        headers: {
+          'Access-Control-Allow-Headers': '*',
+          'Access-Control-Allow-Origin': '*',
+        },
+      });
+    } else if (empty) {
+      await route.fulfill({
+        status: 200,
+        contentType: 'application/json',
+        body: '{"Forms": []}',
+      });
+    } else {
+      await route.fulfill({
+        path: './tests/json/formList.json',
+      });
+    }
+  });
+}
+
+export async function mockFormsFormID(page: page, formStatus: number) {
+  // clear current mock
+  await page.unroute(`${process.env.DELA_PROXY_URL}/evoting/forms/${FORMID}`);
+  await page.route(`${process.env.DELA_PROXY_URL}/evoting/forms/${FORMID}`, async (route) => {
+    let formFile = '';
+    switch (formStatus) {
+      case 0:
+        formFile = 'created.json';
+        break;
+      case 1:
+        formFile = 'open.json';
+        break;
+      case 2:
+        formFile = 'closed.json';
+        break;
+      case 3:
+        formFile = 'shuffled.json';
+        break;
+      case 4:
+        formFile = 'decrypted.json';
+        break;
+      case 5:
+        formFile = 'combined.json';
+        break;
+    }
+    await route.fulfill({
+      path: `./tests/json/evoting/forms/${formFile}`,
+    });
+  });
+}
+
+export async function mockDKGActors(page: page, dkgActorsStatus: number, initialized: boolean) {
+  for (const worker of [Worker0, Worker1, Worker2, Worker3]) {
+    await page.route(`${worker.Proxy}/evoting/services/dkg/actors/${FORMID}`, async (route) => {
+      if (route.request().method() === 'PUT') {
+        await route.fulfill({
+          status: 200,
+          headers: {
+            'Access-Control-Allow-Headers': '*',
+            'Access-Control-Allow-Origin': '*',
+          },
+        });
+      } else {
+        let dkgActorsFile = '';
+        switch (dkgActorsStatus) {
+          case 0:
+            dkgActorsFile = initialized ? 'initialized.json' : 'uninitialized.json';
+            break;
+          //          case 1:
+          //            dkgActorsFile = 'setup.json';
+          //            break;
+          case 6:
+            dkgActorsFile = worker === Worker0 ? 'setup.json' : 'certified.json'; // one node is set up, the remaining nodes are certified
+            break;
+        }
```

Comment:
Does this need a `default` case?

Reply:
no, there are only the three states, and since the tests need specific states I think it's better to pass them explicitely